keel-ec2-api/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/SecurityGroupRule.kt

```diff
@@ -18,9 +18,10 @@ package com.netflix.spinnaker.keel.api.ec2
 import com.netflix.spinnaker.keel.api.schema.Literal
 import com.netflix.spinnaker.keel.api.schema.Optional
 
-sealed class SecurityGroupRule {
+abstract class SecurityGroupRule {
```

Comment:
@robfletcher I have a suspicion that this change is what's causing this kind of error in the API schema tests:
```
  org.opentest4j.AssertionFailedError: ▼ Expect that validation messages for security-group-example.yml:
    ✗ is valid
      found 2 validation errors: [$.environments[0].resources[0].spec.inboundRules[0].protocol: is not defined in the schema and the schema does not allow additional properties, $.environments[0].resources[0].spec.inboundRules[0].portRange: is not defined in the schema and the schema does not allow additional properties]
```
Any ideas?

Reply:
Confirmed: if I change this back to `sealed`, the tests pass. Not sure how to proceed here.

Reply:
Fixed by manually registering the sub-types of `SecurityGroupRule` with the `ExtensionRegistry`.